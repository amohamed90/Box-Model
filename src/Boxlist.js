import React, {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';

function Boxlist() {
  const [boxlist, setBoxlist] = useState([]);
  const add = boxObj => setBoxlist(boxlist => [...boxlist, boxObj]);
  const remove = function(id) {
    setBoxlist(boxlist => boxlist.filter(box => box.id !== id));
  };
  const boxComponents = boxlist.map(box => (
    <Box
      key={box.id}
      id={box.id}
      width={box.width} 
      height={box.height} 
      backgroundColor={box.backgroundColor} 
      removeBox={remove}
    />
    )
  )

  return (
    <div>
      <div className="Boxlist" data-testid="Boxlist">
        {boxComponents}
      </div>
      <NewBoxForm createBox={add}/>
    </div>

  )
}
export default Boxlist;

