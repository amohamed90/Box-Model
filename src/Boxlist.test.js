import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Boxlist from "./Boxlist";

function addBox(boxlist, height = "2", width = "2", color = "peachpuff") {
  const heightInput = boxlist.getByLabelText("HEIGHT");
  const widthInput = boxlist.getByLabelText("WIDTH");
  const backgroundInput = boxlist.getByLabelText("BACKGROUND COLOR");
  fireEvent.change(backgroundInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxlist.getByText("Create Box");
  fireEvent.click(button);
}

// smoke test
it("renders without crashing", function () {
  render(<Boxlist />);
});

// snapshot test
it("matches snapshot", function () {
  const boxlist = render(<Boxlist />);
  expect(boxlist.asFragment()).toMatchSnapshot();
});

// test that add box works
it("can add a box to the BoxList", function () {
  const boxlist = render(<Boxlist />);

  expect(boxlist.getByTestId("Boxlist")).toBeEmpty();
  addBox(boxlist);
  expect(boxlist.getByTestId("Boxlist")).not.toBeEmpty();
  //check for the remove button
  //check for style
  const removeButton = boxlist.getByText("X");
  expect(removeButton).toBeInTheDocument();
  expect(boxlist.getByText("X").closest("div")).toHaveStyle(`
    width: 2px;
    height: 2px;
    background-color: peachpuff;
  `);
});

// test that delete box works
it("can delete a box to the BoxList", function () {
  const boxlist = render(<Boxlist />);

  expect(boxlist.getByTestId("Boxlist")).toBeEmpty();
  addBox(boxlist);
  const removeButton = boxlist.getByText("X");
  fireEvent.click(removeButton);
  expect(boxlist.getByTestId("Boxlist")).toBeEmpty();
});