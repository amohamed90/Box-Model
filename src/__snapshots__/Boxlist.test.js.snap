// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Boxlist"
      data-testid="Boxlist"
    />
    <div
      class="NewBoxForm"
    >
      <form>
        <label
          for="height"
        >
          HEIGHT
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          WIDTH
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
        <label
          for="backgroundColor"
        >
          BACKGROUND COLOR
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
        <input
          type="submit"
          value="Create Box"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
